src/errors.rs

```diff
@@ -114,6 +114,63 @@ impl From<serde_json::Error> for PruneError {
     }
 }
 
+/// The errors that can be returned from [crate::sync::mount]
+#[derive(Debug)]
+pub enum MountError {
+    /// An unknown error occurred
+    Unknown(String),
+    /// Error while splitting the arguments
+    ShlexError,
+    /// The command failed to execute
+    CommandFailed(io::Error),
+    /// Invalid borg output found
+    InvalidBorgOutput(io::Error),
+    /// Error while deserializing output of borg
+    DeserializeError(serde_json::Error),
+    /// Borg was terminated by a signal
+    TerminatedBySignal,
+    /// Piping from stdout or stderr failed
+    PipeFailed,
+    /// The specified archive name already exists
+    ArchiveAlreadyExists,
```

Comment:
I don't think these variants are actually used (I also have missed those in the `PruneError`, so no blame on you)

Reply:
Another copy/paste error, fixed. It seems plausible the other variants could be instantiated so I left them

Reply:
Added `MountSource`, lemme know if you like the idea. I'm happy with whatever though.

Reply:
> Added `MountSource`, lemme know if you like the idea. I'm happy with whatever though.

LGTM

Reply:
Sounds good, I'll resolve this thread then